drivers/console/ipm_console.c

```diff
@@ -46,26 +96,38 @@ static void ipm_console_hook_install(void)
 
 static int ipm_console_init(struct device *dev)
 {
+	struct device *ipm;
+
 	ARG_UNUSED(dev);
 
-	LOG_DBG("");
+	LOG_DBG("IPM console initialization");
 
-	ipm_console_device = device_get_binding(CONFIG_IPM_CONSOLE_ON_DEV_NAME);
-	if (!ipm_console_device) {
+	ipm = device_get_binding(CONFIG_IPM_CONSOLE_ON_DEV_NAME);
+	if (!ipm) {
 		LOG_ERR("Cannot get %s", CONFIG_IPM_CONSOLE_ON_DEV_NAME);
 		return -ENODEV;
 	}
 
+	if (ipm_max_id_val_get(ipm) < 0xFF) {
+		LOG_ERR("IPM driver does not support 8 bit ID values");
+		return -ENOTSUP;
+	}
+
+#if defined(CONFIG_IPM_CONSOLE_IMMEDIATE)
+	ipm_dev = ipm;
+#else
+	k_thread_create(&thread_data, thread_stack,
+			K_THREAD_STACK_SIZEOF(thread_stack),
+			msgq_thread, ipm, NULL, NULL,
+			CONFIG_MAIN_THREAD_PRIORITY - 1,
```

Comment:
If this driver is carrying console output (!), I can pretty much guarantee that "one higher than the app main thread priority" is the wrong value to hard code.

I mean.... what happens when someone spins in a high priority thread and blocks console output?  The system deadlocks silently, right?